service/storage_service.cc

```diff
@@ -4661,8 +4661,15 @@ future<> storage_service::raft_check_and_repair_cdc_streams() {
         }
 
         curr_gen = _topology_state_machine._topology.current_cdc_generation_id;
-
-        // FIXME: check if the current generation is optimal, don't request new one if it isn't
+        if (!curr_gen) {
+            slogger.error("check_and_repair_cdc_streams: no current CDC generation, requesting a new one.");
+        }
+        else {
+            auto gen = co_await _sys_ks.local().read_cdc_generation(curr_gen->id);
+            if (cdc::is_cdc_generation_optimal(gen, get_token_metadata())) {
+                co_return;
```

Comment:
Please print an INFO message.

Reply:
I added an INFO message to both cases - the generation is optimal and it is not.